### ISSUE TITLE
E117: Unknown function: JobHandler

### ISSUE THREAD

**Issue by s-ol** [djmoch/vim-makejob#1]
Installed using Vundle:

    nmap <Leader>e :MakeJob<CR>
    nmap <Leader>E :cwindow<CR>
    Plugin 'djmoch/vim-makejob'

when I run it using `:MakeJob` or using the mapped key, it says:

    make started
    E117: Unknown function: JobHandler

(make is `makeprg`)

Also it's a shame autocmds don't work, couldn't there at least be an option to open the quickfix window if there are errors?

**Comment by djmoch** [comment]
I'll start with your second question. Hooking into the autocmds is the very next thing I plan to do. I'm trying to keep MakeJob a drop-in replacement for make and lmake, so this is a no-brainer. I'm guessing you're using something like `autocmd QuickFixCmdPost * copen`?

Now for your issue, it looks like `s:Function` failed to find an `<SNR>#` match for `JobHandler`. The right answer might be to convert the handler to an `autoload` function, but since this works for me on macOS and Windows, I have a couple questions:

1. What patch level of Vim 8 are you using?
2. Are you on an Linux environment, or something other than macOS or Windows?
3. Do you have tpope's [Scriptease](https://github.com/tpope/vim-scriptease) installed? If so, would you mind opening to line 19 of makejob.vim and running `:Breakadd here`, then `:breakl` and reply back with the exact name of the function? I'm looking for something unusual about the way it's named.

**Comment by djmoch** [comment]
FYI, #2 will deal with the autocmd feature.

**Comment by s-ol** [comment]
from version:

```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Oct 30 2016 15:27:09)
Included patches: 1-55
Compiled by Arch Linux
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl             +clipboard       +dialog_con_gui  +file_in_path    +job             +lua/dyn         +mouse_sgr       +path_extra      +rightleft       +tag_old_static  +user_commands   +writebackup
+arabic          +cmdline_compl   +diff            +find_in_path    +jumplist        +menu            -mouse_sysmouse  +perl/dyn        +ruby/dyn        -tag_any_white   +vertsplit       +X11
+autocmd         +cmdline_hist    +digraphs        +float           +keymap          +mksession       +mouse_urxvt     +persistent_undo +scrollbind      +tcl/dyn         +virtualedit     -xfontset
+balloon_eval    +cmdline_info    +dnd             +folding         +lambda          +modify_fname    +mouse_xterm     +postscript      +signs           +termguicolors   +visual          +xim
+browse          +comments        -ebcdic          -footer          +langmap         +mouse           +multi_byte      +printer         +smartindent     +terminfo        +visualextra     -xpm
++builtin_terms  +conceal         +emacs_tags      +fork()          +libcall         +mouseshape      +multi_lang      +profile         +startuptime     +termresponse    +viminfo         +xsmp_interact
+byte_offset     +cryptv          +eval            +gettext         +linebreak       +mouse_dec       -mzscheme        +python/dyn      +statusline      +textobjects     +vreplace        +xterm_clipboard
+channel         +cscope          +ex_extra        -hangul_input    +lispindent      +mouse_gpm       +netbeans_intg   +python3/dyn     -sun_workshop    +timers          +wildignore      -xterm_save
+cindent         +cursorbind      +extra_search    +iconv           +listcmds        -mouse_jsbterm   +num64           +quickfix        +syntax          +title           +wildmenu
+clientserver    +cursorshape     +farsi           +insert_expand   +localmap        +mouse_netterm   +packages        +reltime         +tag_binary      +toolbar         +windows
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I
/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/in
clude -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -D_FORTIFY_SOURCE=2  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -U
_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro -L/usr/local/lib
 -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lSM -lICE -lXt -lX11 -lXd
mcp -lSM -lICE  -lm -lncurses -lelf -lnsl    -lacl -lattr -lgpm -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/pe
rl5/core_perl/CORE -lperl -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc   -L/usr/lib -ltclstub8.6 -ldl -lz -lpthread -lieee -lm
```
on arch linux, gvim package from repos.

I don't have scriptease but i'll give it a shot

**Comment by s-ol** [comment]
It's called `func <SNR>26_JobHandler`

**Comment by djmoch** [comment]
Fixed by #4 in 9f73945445bbe11f6da0e45cc256622220d57e17
